tests/models/test_taskinstance.py

```diff
@@ -537,10 +519,10 @@ def reschedule():
         session.commit()
 
         ti.task = task
-        if state in [State.FAILED, State.UP_FOR_RETRY]:
+        if exception_type == AirflowException:
```

Comment:
This does not correctly cover the `AirflowRescheduleException(timezone.utcnow())` case. It’s probably better to change `exception_type` to `exception` instead and always instantiate to avoid this subtle issue caused by `exception_type` may be either a type or an instance.

Reply:
Thanks for the feedback, I will rename `exception_type` to `exception`. I will edit as follows:

```
if state in [State.FAILED, State.UP_FOR_RETRY]:
    with pytest.raises(exception):
        ti.run()
else:
    ti.run()
```

This ensures that the `exception` is tied to whatever `state` is being checked. 

The reason I did not include `AirflowRescheduleException(timezone.utcnow())` is because that exception is caught during `ti.run()`.